src/clj/vanilla/db/core.clj

```diff
@@ -95,29 +89,75 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; INITIAL DB SETUP FUNCTION
 ;;
-;; not called anywhere in the app
 ;; run in the repl to create fresh db tables populated accordingly
 ;;
 ;;;;;;;;;;;;;;;;;;;;;;;;;;
-(defn initialize-database []
+(defn initialize-database
+  [database]
   (do
     ;; Remove any current tables to start fresh
-    (drop-services-table vanilla-db)
-    (drop-layout-table vanilla-db)
-    (drop-users-table vanilla-db)
+    (drop-services-table database)
+    (drop-layout-table database)
+    (drop-users-table database)
 
     ;; Create new empty versions of the tables
-    (create-services-table vanilla-db)
-    (create-layout-table vanilla-db)
-    (create-user-table vanilla-db)
+    (create-services-table database)
+    (create-layout-table database)
+    (create-user-table database)
 
     ;; Bootstrap any data needed in the DB at start
-    (populate-services)))
+    (populate-services database)))
+
+
+(defrecord Database [db-spec                                ; configuration
+                     datasource]                            ; state
+
+  component/Lifecycle
+  (start [this]
+    (if datasource
+      this                                                  ; already initialized
+      (assoc this :datasource (jdbc/get-datasource db-spec))))
+
+  (stop [this]
+    (assoc this :datasource nil)))
+
 
-;;; REPL ME vvvvv ;;;
+(defn database-exist?
+  "Runs a simple check to see if the vanilla_db file exists."
+  [database]
+  ;(prn "Database-exist? /////" (database :dbname))
+  ;(prn (str (.exists (jio/file "vanilla_db"))))
+  (.exists (jio/file (database :dbname))))
+
+
+
+(defn setup-database
+  "This is called by dashboard-clj.system to start the applications
+   connection to the database. Only creates a database when in dev mode
+   and when the vanilla_db has not been created."
+  [dev-mode?]
+  (when (and (not (database-exist? vanilla-db)) dev-mode?)
+    ;; If it does not exist, create it
+    (initialize-database vanilla-db)
+    (log/info "creating startup database: VANILLA_DB"))
+
+  ;; Create a Database component -- see component library
+  (map->Database {:db-spec vanilla-db}))
+
+
+
+;;; REPL ME WHEN DATABASE STRUCTURE CHANGES ;;;
 (comment
 
-  (initialize-database)
+  ;; The following is the deployed database that is pushed to the repo and to AWS
+  ;; It is not called by any of our production code, so it is in this comment block
+  (def vanilla-default
+    "SQLite database connection spec."
+    {:dbtype db-type :dbname "vanilla_default"})
+
+  (initialize-database vanilla-default)                     ;; Initialize this on database structure changes, push changes to repo
```

Comment:
@bfhub Here is the call, it's in a similar spot as before just everything around it moved.